### PR TITLE
DDF-1855 Fixed csw xpath queries with namespaces

### PR DESCRIPTION
**Pull request by bcwaters**
Updated the DefaultCswRecordMap so that xpath queries remain
unchanged when checking for namespace prefixes.  Unit tests have
been included to demonstrate the intended functionality is
implemented.

@tbatie @ryeats @roelens8 @rzwiefel @kcwire

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/codice/ddf/pull/527" target="_blank">ReviewNinja

**Comment by tbatie** [comment]
:+1:

File: catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/DefaultCswRecordMap.java [hunk 1]
```diff
@@ -215,6 +217,10 @@ public static boolean hasDefaultMetacardFieldForPrefixedString(String propertyNa
         }
     }
 
+    private static boolean isXpathPropertyName(String propertyName) {
+        return ((propertyName.indexOf('/') != -1 || propertyName.indexOf('@') != -1));
```
Comment (rzwiefel) [inline catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/DefaultCswRecordMap.java]: Why not use propertyName.contains?

Comment (rzwiefel) [inline catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/DefaultCswRecordMap.java]: And you shouldn't need parentheses around it

**Comment by ryeats** [comment]
:+1:

**Comment by roelens8** [comment]
:+1:

File: catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/DefaultCswRecordMap.java (hunk 1 above)
Comment (bcwaters) [inline catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/DefaultCswRecordMap.java]: You're right, that would help readability. I'll update the code.

Comment (tbatie) [inline catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/DefaultCswRecordMap.java]: I thought we should only assume it is xpath if it begins with a '/' or '@' so shouldn't we only be checking the first index?

**Comment by roelens8** [comment]
Hero'ed successfully! :shipit:

**Pull request merged**